packages/block-editor/src/components/block-list/index.js

```diff
@@ -135,10 +135,31 @@ function Items( {
 				getSelectedBlockClientIds,
 				__unstableGetVisibleBlocks,
 			} = select( blockEditorStore );
+			const _order = getBlockOrder( rootClientId );
+			const _selectedBlocks = getSelectedBlockClientIds();
+			const _visibleBlocks = __unstableGetVisibleBlocks();
+			let hasSelectedBlocks = false;
+			let hasVisibleBlocks = false;
+			for ( const clientId of _order ) {
+				if (
+					! hasSelectedBlocks &&
+					_selectedBlocks.includes( clientId )
+				) {
+					hasSelectedBlocks = true;
+				}
+
+				if ( ! hasVisibleBlocks && _visibleBlocks.has( clientId ) ) {
+					hasVisibleBlocks = true;
+				}
+
+				if ( hasSelectedBlocks && hasVisibleBlocks ) {
+					break;
+				}
+			}
```

Comment:
Could this be a private selectors, something like `getSelectedBlocksAtLevel( rootClientId )` and `getVisibleBlocksAtLevel( rootClientID )` or something like that?

Reply:
I was thinking about that, but it didn't feel worth it to create another selector for this, especially since it's such a weird optimisation. Also we already have access to the block order here. What do you think?

Reply:
Ideally we also prevent re-rendering the current list with the selected block, but I don't know of a way to do that without introducing another block editor store subscription _per block_.

Reply:
To be honest, with private APIs, I think the purpose becomes clearer with a selector that without.

Reply:
Ok, will add them as selectors then :)